### PR TITLE
networkd: vlan add GVRP support

### LINKED ISSUE #5760: RFE: GVRP support for VLANs in systemd-networkd
### Submission type

  - [ ] Bug report
  - [x] Request for enhancement (RFE)

### GVRP Support in systemd-networkd

When a VLAN is started by systemd-networkd, I would like to control whether to use GVRP or not. I request to add an option in the .netdev file under the [VLAN] tag. An option like Gvrp=yes/no should enable or disable the use of GVRP.

Of course there are also other options possible, like MVRP

### Example

```
[VLAN]
Id=100
Gvrp=yes
```

`Gvrp=yes` &rarr; ip link add link _dev_.100 _dev_ type vlan id 100 **gvrp on**
`Gvrp=no` &rarr; ip link add link _dev_.100 _dev_ type vlan id 100 **gvrp off**

### PR DESCRIPTION
**Pull request by ssahani**
Add support to configure GVRP.

Closes #5760

### CODE REVIEW

File: src/network/netdev/vlan.c [hunk 1]
```diff
@@ -17,12 +17,14 @@
...
 ***/
 
+#include <linux/if_vlan.h>
 #include <net/if.h>
 
 #include "netdev/vlan.h"
 #include "vlan-util.h"
 
 static int netdev_vlan_fill_message_create(NetDev *netdev, Link *link, sd_netlink_message *req) {
+        struct ifla_vlan_flags flags = { 0 };
```
Comment (keszybz) [inline src/network/netdev/vlan.c]: `{}`

File: src/network/netdev/vlan.c [hunk 2]
```diff
@@ -38,6 +40,20 @@ static int netdev_vlan_fill_message_create(NetDev *netdev, Link *link, sd_netlin
         if (r < 0)
                 return log_netdev_error_errno(netdev, r, "Could not append IFLA_VLAN_ID attribute: %m");
 
+        if (v->gvrp != -1) {
+
```
Comment (keszybz) [inline src/network/netdev/vlan.c]: Spurious newline.

File: man/systemd.netdev.xml [hunk 3]
```diff
@@ -399,8 +399,14 @@
             This option is compulsory.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>GVRP=</varname></term>
+          <listitem>
+            <para>The Generic VLAN Registration Protocol (GVRP) is a protocol that
+            allows automatic learning of VLANs on a network. A boolean. Defaults to unset.</para>
```
Comment (keszybz) [inline man/systemd.netdev.xml]: What does "unset" mean here?

**Comment by ssahani** [comment]
updated .

**Comment by Aison0** [comment]
Thx :-) Maybe one can think about adding MVRP support also. The implementation would be almost similar.

File: man/systemd.netdev.xml (hunk 3 above)
Comment (Aison0) [reply man/systemd.netdev.xml]: Maybe it should be changed to "Defaults to system settings"? 
true=enabled
false=disabled
unset=System Default?

**Pull request merged**